src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheck.java

```diff
@@ -371,6 +380,7 @@ public int[] getAcceptableTokens() {
             TokenTypes.RECORD_DEF,
             TokenTypes.COMPACT_CTOR_DEF,
             TokenTypes.LITERAL_SWITCH,
+            TokenTypes.VARIABLE_DEF,
```

Comment:
@nrmancuso @romani 

for an input file like this:
```java
/*
RightCurly
option = ALONE_OR_SINGLELINE
tokens = LITERAL_TRY , LITERAL_CATCH , LITERAL_FINALLY , LITERAL_IF , LITERAL_ELSE , CLASS_DEF , \
         METHOD_DEF , CTOR_DEF , LITERAL_FOR , LITERAL_WHILE , LITERAL_DO , STATIC_INIT , \
         INSTANCE_INIT , ANNOTATION_DEF , ENUM_DEF , INTERFACE_DEF , RECORD_DEF , COMPACT_CTOR_DEF \
         , VARIABLE_DEF
*/

package com.puppycrawl.tools.checkstyle.checks.blocks.rightcurly;

public class InputRightCurlyFollowedBySemicolonHasLineBreakAfter {


        Object obj2_normal = new Object();

        Object obj_anon = new Object() {};

        Object obj_anon = new Object() {

        public void method_in_anon_class() {};

        };

        Object obj_anon = new Object() {

        class inner_class{};

        };

        Object obj_anon2 = new Object() {

        Object obj_inner_anon2 = new Object() {

        };

        };
}
```

The generated AST is:
```
COMPILATION_UNIT -> COMPILATION_UNIT [12:0]
|--PACKAGE_DEF -> package [12:0]
|   |--ANNOTATIONS -> ANNOTATIONS [12:53]
|   |--DOT -> . [12:53]
|   |   |--DOT -> . [12:46]
|   |   |   |--DOT -> . [12:39]
|   |   |   |   |--DOT -> . [12:28]
|   |   |   |   |   |--DOT -> . [12:22]
|   |   |   |   |   |   |--DOT -> . [12:11]
|   |   |   |   |   |   |   |--IDENT -> com [12:8]
|   |   |   |   |   |   |   `--IDENT -> puppycrawl [12:12]
|   |   |   |   |   |   `--IDENT -> tools [12:23]
|   |   |   |   |   `--IDENT -> checkstyle [12:29]
|   |   |   |   `--IDENT -> checks [12:40]
|   |   |   `--IDENT -> blocks [12:47]
|   |   `--IDENT -> rightcurly [12:54]
|   `--SEMI -> ; [12:64]
`--CLASS_DEF -> CLASS_DEF [14:0]
    |--MODIFIERS -> MODIFIERS [14:0]
    |   `--LITERAL_PUBLIC -> public [14:0]
    |--LITERAL_CLASS -> class [14:7]
    |--IDENT -> InputRightCurlyFollowedBySemicolonHasLineBreakAfter [14:13]
    `--OBJBLOCK -> OBJBLOCK [14:65]
        |--LCURLY -> { [14:65]
        |--VARIABLE_DEF -> VARIABLE_DEF [17:8]
        |   |--MODIFIERS -> MODIFIERS [17:8]
        |   |--TYPE -> TYPE [17:8]
        |   |   `--IDENT -> Object [17:8]
        |   |--IDENT -> obj2_normal [17:15]
        |   |--ASSIGN -> = [17:27]
        |   |   `--EXPR -> EXPR [17:29]
        |   |       `--LITERAL_NEW -> new [17:29]
        |   |           |--IDENT -> Object [17:33]
        |   |           |--LPAREN -> ( [17:39]
        |   |           |--ELIST -> ELIST [17:40]
        |   |           `--RPAREN -> ) [17:40]
        |   `--SEMI -> ; [17:41]
        |--VARIABLE_DEF -> VARIABLE_DEF [19:8]
        |   |--MODIFIERS -> MODIFIERS [19:8]
        |   |--TYPE -> TYPE [19:8]
        |   |   `--IDENT -> Object [19:8]
        |   |--IDENT -> obj_anon [19:15]
        |   |--ASSIGN -> = [19:24]
        |   |   `--EXPR -> EXPR [19:26]
        |   |       `--LITERAL_NEW -> new [19:26]
        |   |           |--IDENT -> Object [19:30]
        |   |           |--LPAREN -> ( [19:36]
        |   |           |--ELIST -> ELIST [19:37]
        |   |           |--RPAREN -> ) [19:37]
        |   |           `--OBJBLOCK -> OBJBLOCK [19:39]
        |   |               |--LCURLY -> { [19:39]
        |   |               `--RCURLY -> } [19:40]
        |   `--SEMI -> ; [19:41]
        |--VARIABLE_DEF -> VARIABLE_DEF [21:8]
        |   |--MODIFIERS -> MODIFIERS [21:8]
        |   |--TYPE -> TYPE [21:8]
        |   |   `--IDENT -> Object [21:8]
        |   |--IDENT -> obj_anon [21:15]
        |   |--ASSIGN -> = [21:24]
        |   |   `--EXPR -> EXPR [21:26]
        |   |       `--LITERAL_NEW -> new [21:26]
        |   |           |--IDENT -> Object [21:30]
        |   |           |--LPAREN -> ( [21:36]
        |   |           |--ELIST -> ELIST [21:37]
        |   |           |--RPAREN -> ) [21:37]
        |   |           `--OBJBLOCK -> OBJBLOCK [21:39]
        |   |               |--LCURLY -> { [21:39]
        |   |               |--METHOD_DEF -> METHOD_DEF [23:8]
        |   |               |   |--MODIFIERS -> MODIFIERS [23:8]
        |   |               |   |   `--LITERAL_PUBLIC -> public [23:8]
        |   |               |   |--TYPE -> TYPE [23:15]
        |   |               |   |   `--LITERAL_VOID -> void [23:15]
        |   |               |   |--IDENT -> method_in_anon_class [23:20]
        |   |               |   |--LPAREN -> ( [23:40]
        |   |               |   |--PARAMETERS -> PARAMETERS [23:41]
        |   |               |   |--RPAREN -> ) [23:41]
        |   |               |   `--SLIST -> { [23:43]
        |   |               |       `--RCURLY -> } [23:44]
        |   |               |--SEMI -> ; [23:45]
        |   |               `--RCURLY -> } [25:8]
        |   `--SEMI -> ; [25:9]
        |--VARIABLE_DEF -> VARIABLE_DEF [27:8]
        |   |--MODIFIERS -> MODIFIERS [27:8]
        |   |--TYPE -> TYPE [27:8]
        |   |   `--IDENT -> Object [27:8]
        |   |--IDENT -> obj_anon [27:15]
        |   |--ASSIGN -> = [27:24]
        |   |   `--EXPR -> EXPR [27:26]
        |   |       `--LITERAL_NEW -> new [27:26]
        |   |           |--IDENT -> Object [27:30]
        |   |           |--LPAREN -> ( [27:36]
        |   |           |--ELIST -> ELIST [27:37]
        |   |           |--RPAREN -> ) [27:37]
        |   |           `--OBJBLOCK -> OBJBLOCK [27:39]
        |   |               |--LCURLY -> { [27:39]
        |   |               |--CLASS_DEF -> CLASS_DEF [29:8]
        |   |               |   |--MODIFIERS -> MODIFIERS [29:8]
        |   |               |   |--LITERAL_CLASS -> class [29:8]
        |   |               |   |--IDENT -> inner_class [29:14]
        |   |               |   `--OBJBLOCK -> OBJBLOCK [29:25]
        |   |               |       |--LCURLY -> { [29:25]
        |   |               |       `--RCURLY -> } [29:26]
        |   |               |--SEMI -> ; [29:27]
        |   |               `--RCURLY -> } [31:8]
        |   `--SEMI -> ; [31:9]
        |--VARIABLE_DEF -> VARIABLE_DEF [33:8]
        |   |--MODIFIERS -> MODIFIERS [33:8]
        |   |--TYPE -> TYPE [33:8]
        |   |   `--IDENT -> Object [33:8]
        |   |--IDENT -> obj_anon2 [33:15]
        |   |--ASSIGN -> = [33:25]
        |   |   `--EXPR -> EXPR [33:27]
        |   |       `--LITERAL_NEW -> new [33:27]
        |   |           |--IDENT -> Object [33:31]
        |   |           |--LPAREN -> ( [33:37]
        |   |           |--ELIST -> ELIST [33:38]
        |   |           |--RPAREN -> ) [33:38]
        |   |           `--OBJBLOCK -> OBJBLOCK [33:40]
        |   |               |--LCURLY -> { [33:40]
        |   |               |--VARIABLE_DEF -> VARIABLE_DEF [35:8]
        |   |               |   |--MODIFIERS -> MODIFIERS [35:8]
        |   |               |   |--TYPE -> TYPE [35:8]
        |   |               |   |   `--IDENT -> Object [35:8]
        |   |               |   |--IDENT -> obj_inner_anon2 [35:15]
        |   |               |   |--ASSIGN -> = [35:31]
        |   |               |   |   `--EXPR -> EXPR [35:33]
        |   |               |   |       `--LITERAL_NEW -> new [35:33]
        |   |               |   |           |--IDENT -> Object [35:37]
        |   |               |   |           |--LPAREN -> ( [35:43]
        |   |               |   |           |--ELIST -> ELIST [35:44]
        |   |               |   |           |--RPAREN -> ) [35:44]
        |   |               |   |           `--OBJBLOCK -> OBJBLOCK [35:46]
        |   |               |   |               |--LCURLY -> { [35:46]
        |   |               |   |               `--RCURLY -> } [37:8]
        |   |               |   `--SEMI -> ; [37:9]
        |   |               `--RCURLY -> } [39:8]
        |   `--SEMI -> ; [39:9]
        `--RCURLY -> } [42:0]
```

If you notice, I will have to provide support for variable def so that I can trace all the way down to OBJ_BLOCK which will help me identify the anonymous inner class.